juju/python-libjuju
# [JUJU-2276] Series or base for local charms

#### Description

This adds to the pylibjuju the capability to check also the `charmcraft.yaml` to gather information about series/base of a local charm at deploy. Without this, operators (like [grafana](https://github.com/canonical/grafana-k8s-operator) for instance) are having trouble deploying when running their own integration tests.

Should fix #765 

#### QA Steps

Adding a new test with manually crafted local charm (at `tests/integration/charm-base-charmcraft-yaml`) that doesn't have a series info in the metadata.yaml, but it does have a `charmcraft.yaml` that has:

```yaml
bases:
  - build-on:
      - name: "ubuntu"
        channel: "20.04"
    run-on:
      - name: "ubuntu"
        channel: "20.04"
```

the test deploys this charm, so the following should pass:
```
tox -e integration -- tests/integration/test_model.py::test_deploy_local_charm_base_charmcraft_yaml
```

There's also a second way to QA this with cloning the https://github.com/canonical/grafana-k8s-operator and trying to deploy that one using pylibjuju. I modified the `examples/localcharm.py` to deploy the grafana from its directory and it worked well :+1: 

#### Notes & Discussion

I tried this on the `2.9.36` controller as the reported bug was using it, it might also be beneficial to test it on a `3.0` controller.
